### PR TITLE
Tidy up validate-modules:parameter-list-no-elements (batch 1)

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Fixed missing `elements` in both docs and `argument_spec` for a bunch of modules.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/doc_fragments/oracle_creatable_resource.py
plugins/module_utils/oracle/oci_utils.py
plugins/modules/cloud/lxd/lxd_container.py
plugins/modules/cloud/packet/packet_device.py
plugins/modules/cloud/pubnub/pubnub_blocks.py
plugins/modules/cloud/smartos/vmadm.py
plugins/modules/cloud/softlayer/sl_vm.py
plugins/modules/files/xml.py
plugins/modules/identity/keycloak/keycloak_client.py
plugins/modules/identity/onepassword_info.py
plugins/modules/monitoring/librato_annotation.py
plugins/modules/monitoring/pagerduty.py
plugins/modules/net_tools/dnsimple.py
plugins/modules/net_tools/nsupdate.py
plugins/modules/net_tools/omapi_host.py
plugins/modules/notification/mail.py
plugins/modules/notification/nexmo.py
plugins/modules/notification/office_365_connector_card.py
plugins/modules/notification/rocketchat.py
plugins/modules/notification/sendgrid.py
plugins/modules/notification/slack.py
plugins/modules/notification/twilio.py
plugins/modules/packaging/os/redhat_subscription.py
plugins/modules/source_control/gitlab/gitlab_runner.py
plugins/modules/storage/netapp/na_ontap_gather_facts.py

### CODE REVIEW

**Comment by russoz** [comment]
changelog frag coming up as soon as I get the checks figured out (before some @felixfontein point that out to me :P )

**Comment by russoz** [comment]
@felixfontein done here, hit it :-)

File: changelogs/fragments/1795-list-elements-batch1.yml
Comment (felixfontein) [inline changelogs/fragments/1795-list-elements-batch1.yml:2]:
```suggestion
  - oci_utils module_utils - elements of list parameter ``key_by`` are now validated (https://github.com/ansible-collections/community.general/pull/1795).
```
(The `is` → `are` also for the other entries.)

File: plugins/modules/cloud/smartos/vmadm.py
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:79]: I'd really wish the documentation here would be slightly more helpful.

File: changelogs/fragments/1795-list-elements-batch1.yml
Comment (russoz) [reply changelogs/fragments/1795-list-elements-batch1.yml:2]: instead of that, maybe:
>  `elements` of list param `xxx` is
?

Comment (russoz) [reply changelogs/fragments/1795-list-elements-batch1.yml:2]: or:
> the `elements` setting of the list params xxx is

Comment (felixfontein) [reply changelogs/fragments/1795-list-elements-batch1.yml:2]: I think it should still be `are` then, because elements is plural.

Comment (felixfontein) [reply changelogs/fragments/1795-list-elements-batch1.yml:2]: > > the `elements` setting of the list params xxx is

The `elements` setting is not validated, but the elements of the list option are validated.

Comment (russoz) [reply changelogs/fragments/1795-list-elements-batch1.yml]: fair enough, commit coming up

File: plugins/modules/cloud/smartos/vmadm.py
Comment (russoz) [reply plugins/modules/cloud/smartos/vmadm.py:79]: Couldn't agree more. BOTMETA says this module is owned by $team_solaris. Maybe they can put a more thorough documentation in here?

I found the information to be in:
https://smartos.org/man/1m/vmadm

But I don't think this should be the PR for that.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
I'll keep this open until the next release (beginning of next week) so maintainers have a bit of time to also take a look.

**Comment by russoz** [comment]
That was a little rebasing. Please merge this one after #1766 and #1771 . I got another PR to submit after this one, but that one will require some rebasing before merge. After these guys are all in (hopefully for 3.0.0 next week?), we should have less than a thousand ignore lines in the 3 files combined.

And right now **cloud/ovirt** is the major offender in those files, and if I am seeing it right, it will go away with 3.0.0, so we should see a **very** big drop in the number of lines there.

**Comment by felixfontein** [comment]
I'll merge this one tomorrow. (@russoz looks like a rebase is not necessary.)

**Comment by ndclt** [comment]
@russoz lgtm for keycloak module. Thanks for the work.

**Comment by felixfontein** [comment]
@russoz thanks for fixing these!
@ndctl thanks for reviewing!

**Comment by gaige** [comment]
@russoz Was this patch tested with smartos `vmadm`?  As far as I can tell the changes to `resolvers` have broken `vmadm` with `resolvers` specified in 2.10.7+

**Comment by russoz** [comment]
@gaige It passed the existing unit and integration tests. We did not pursue individual modules' testing. Can you please raise a bug report on this, showing the problem, so I can try and follow up on that? Or if that's a simple change and you're keen of submitting the PR, I'd be more than happy to review.

**Comment by gaige** [comment]
Thanks.

I've raised the bug report; I haven't contributed back to Ansible since the break-out of the modules, so that might not be particularly expedient.  The long and short of the issue is that `resolvers` is an array of strings, and the verifier expects it to be an array of dictionaries.

**Pull request merged**
